Adding docker documentation for the TICK stack

jsternberg opened the pull request:
docker-library/official-images#1583

psftw commented:
Sent over https://github.com/influxdata/docker-docs/pull/2

psftw commented on kapacitor/content.md:
```diff
@@ -0,0 +1,133 @@
+# Kapacitor
+
+Kapacitor is an open source data processing engine written in Go. It can process both stream and batch data.
```
Need a bit more context here

psftw commented on influxdb/content.md:
````diff
@@ -0,0 +1,123 @@
+Generate the default configuration file:
+
+```console
+$ docker run --rm influxdb influxd config > influxdb.conf
````
This doesn't work anymore with the latest changes, but I think it should.

jsternberg replied:
Works again.

psftw commented on kapacitor/content.md:
```diff
@@ -0,0 +1,133 @@
+
+### Configuration
+
+Kapacitor can be either configured from a config file or using environment variables. To mount a configuration file and use it with the server, you can use this command:
```
reword --  "use this command:" followed by a set of instructions/commands

psftw commented on kapacitor/content.md:
````diff
@@ -0,0 +1,133 @@
+Generate the default configuration file:
+
+```console
+$ docker run --rm kapacitor kapacitord config > kapacitor.conf
````
This doesn't work with the latest changes, but I think it should.

jsternberg replied:
Fixed this so it works again.

psftw commented on kapacitor/content.md:
````diff
@@ -0,0 +1,133 @@
+Modify the default configuration, which will now be available under `$PWD`. Then start the Kapacitor container.
+
+```console
+$ docker run -p 9092:9092 \
````
double-check this command

psftw commented on kapacitor/content.md:
```diff
@@ -0,0 +1,133 @@
+
+-	9092 TCP -- HTTP API endpoint
+
+#### Subscriptions
```
I found Kapacitor to be the more confusing component when initially looking at TICK.  I don't find this updated section to be really helpful.  It suggests starting influxdb after Kapacitor which seems backwards, and it also uses older style linking vs. the built in service discovery which is inconsistent with the Telegraf docs.  The syntax for environment variable substitution when lists are involved could be explained as it isn't intuitive yet the URLS setting seems to be pretty important.

jsternberg replied:
When you say built in service discovery, do you mean `--net=container:influxdb` or something else?

tianon replied:
I believe @psftw is referring to `docker network` (which is what is slowly replacing `--link`); see https://docs.docker.com/engine/userguide/networking/dockernetworks/

jsternberg replied:
I've updated the documentation to use `docker network` and remove all references to `--link`.

psftw commented on kapacitor/content.md:
````diff
@@ -0,0 +1,133 @@
+    influxdb
+```
+
+Modify `$HOSTNAME` with a valid hostname for connecting to InfluxDB.
````
Not very helpful instructions?

psftw commented on kapacitor/content.md:
````diff
@@ -0,0 +1,133 @@
+    kapacitor bash
+```
+
+Then from within the container you can use the `kapacitor` command to interact with the daemon. Assuming you have Telegraf + InfluxDB + Kapacitor all hooked up, then create a file `cpu_alert.tick`. If not then see the [docker-compose](https://docs.docker.com/compose/) [TICK stack](https://github.com/influxdata/TICK-docker) environment for easy setup.
````
This link to influxdata/TICK-docker has nothing to do with docker-compose.

psftw commented on kapacitor/content.md:
````diff
@@ -0,0 +1,133 @@
+$ docker run --rm kapacitor kapacitord config > kapacitor.conf
+```
+
+Modify the default configuration, which will now be available under `$PWD`. Then start the Kapacitor container.
````
An example excerpt of a config file as part of the instructions for running kapacitor would be preferred over the later instructions to configure it manually via bash-in-container.

jsternberg commented on influxdb/content.md:
````diff
@@ -0,0 +1,123 @@
+Run the influx client in another container:
+
+```console
+$ docker run --rm --link=influxdb -it influxdb influx -host influxdb
````
This only sometimes works due to a bug in the library that takes care of the line input that I'm fixing in peterh/liner#75 or we'll fork the dependency so we can get the fix. There's also a second bug that causes this to occasionally enter an infinite loop and I'll submit that bugfix to influxdb after the other PR gets merged. We'll definitely have this fix in for 0.13.

jsternberg commented:
I updated the Kapacitor documentation, squashed the commits, and rebased on the latest. I also removed the references to the TICK docker-compose file since it simplified the README.

There's an issue with the influx binary for 0.12 so that won't work properly for now. The issue will be fixed in 0.13, but we would still like it if we could get 0.12 merged in either way.

psftw commented:
Thanks.  Latest changes look good.

tianon commented:
LGTM

yosifkit commented:
LGTM
